documentation/learningPath/testing.md

```diff
@@ -2,3 +2,86 @@
+- [CollectionRuleActions.UnitTests](../../src/Tests/CollectionRuleActions.UnitTests/)
+
+Unit test assemblies directly reference types from various dotnet-monitor assemblies. However, since most of dotnet-monitor heavily relies on code injection, there are utility classes to simplify unit test creation. 
+
+- [TestHostHelper](../../src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTestCommon/TestHostHelper.cs) can be used to setup a basic unit test scenario using dependency injection.
+- [CollectionRuleOptionsExtensions](../../src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTestCommon/Options/CollectionRuleOptionsExtensions.cs) can be used to easily create collection rules from configuration.
+
+## Functional Tests
+
+- [Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests](../../src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests)
+- [Microsoft.Diagnostics.Monitoring.UnitTestApp](../../src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/)
+
+Functional tests are composed of 3 main parts:
+- The test itself, which sets up and validates the results.
+- An instance of dotnet-monitor
+- An instance of an application that is being monitored (from the UnitTestApp assembly)
+
+* [ScenarioRunner](../../src/tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/Runners/ScenarioRunner.cs) is typically used to orchestrate test runs. The class will spawn both an instance of dotnet-monitor and an instance of test application. The app and the test communicate via stdio. The test communicates with dotnet-monitor via its Api surface.
+* The dotnet-monitor Api surface can be accessed through the [ApiClient](../../src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/HttpApi/ApiClient.cs).
+* New scenarios can be added [here](../../src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/).
+* The [AsyncWaitScenario](../../src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/AsyncWaitScenario.cs) is sufficient for most tests.
+
+## Native/Profiler Tests
+
+- [Microsoft.Diagnostics.Monitoring.Profiler](../../src/tests/Microsoft.Diagnostics.Monitoring.Profiler/)
+- [Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp](../../src/tests/Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp/)
+
+This test assembly provides a test to make sure the dotnet-monitor profiler can load into a target app.
+
+## Schema Generation
+
+- [Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests](../../src/tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema.UnitTests/)
+- [Microsoft.Diagnostics.Monitoring.ConfigurationSchema](../../src/tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/)
+- [Microsoft.Diagnostics.Monitoring.Options](../../src/Microsoft.Diagnostics.Monitoring.Options)
+
+Dotnet-monitor generates [schema.json](../../documentation/schema.json) using unit tests. If dotnet-monitor configuration changes, the schema.json file needs to be updated.
```

Comment:
I just remembered that I mention the process for doing this in another part of the learning path in more detail - it might be worth linking to from here https://github.com/wiktork/dotnet-monitor/blob/main/documentation/learningPath/configuration.md#how-configuration-works